### PR TITLE
AMP footer

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

Adds just about enough to render AMP with a footer.

Note, we do not care about breakpoints/media-queries in AMP. The CSS and structure is therefore simpler.

screen shot 2018-10-23 at 13 54 15

Other breakpoints: n/a :)

## Why?

As part of our Milestone 2!

### CODE REVIEW

File: packages/rendering/server.ts [hunk 1]
```diff
@@ -39,10 +41,7 @@ const renderArticle = ({ body }: express.Request, res: express.Response) => {
 
 const renderAMPArticle = ({ body }: express.Request, res: express.Response) => {
     try {
-        const resp = AMPDocument({
-            bodyHTML: '<h1>This is a valid AMP doc!</h1>',
-        });
-
+        const resp = AMPDocument({ body: React.createElement(AMPArticle) });
```
Comment (SiAdcock) [inline packages/rendering/server.ts]: FWIW I think this is a nicer approach than hardcoding the page in the document.

Comment (nicl) [reply packages/rendering/server.ts]: Yeah, only ugliness was having to call React.createElement but doesn't seem too bad.

File: frontend/amp/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        padding: 12px 0 0 10px;
+    }
+`;
+
+const copyright = css`
+    font-size: 12px;
+    padding: 6px 0 18px;
+    border-top: 1px solid #434343;
```
Comment (SiAdcock) [inline frontend/amp/components/Footer.tsx]: Is there a colour in pasteup?

File: frontend/amp/components/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    :hover {
+        text-decoration: underline;
+    }
+`;
+
+const footerList = css`
+    display: flex;
+    flex-wrap: wrap;
+    flex-direction: row;
+    border-top: 1px solid #434343;
```
Comment (AWare) [inline frontend/amp/components/Footer.tsx]: @zeftilldeath what palette colour do we want here?

File: frontend/amp/components/Footer.tsx (hunk 2 above)
Comment (nicl) [reply frontend/amp/components/Footer.tsx]: Previously this was palette 20 (neutral) but it seems to have changed - and also broken PROD guui at somepoint.

I think @AWare and I are planning on asking @zeftilldeath about this.

**Comment by SiAdcock** [comment]
How do you feel about mirroring the `frontend/pages` directory under `amp`? This helps keep the pages together to make it easier for the server to dynamically  discover and render the right page.

File: packages/rendering/server.ts (hunk 1 above)
Comment (SiAdcock) [reply packages/rendering/server.ts]: We could rename this `server.tsx` to give us JSX support?

File: frontend/amp/document.tsx [hunk 4]
```diff
@@ -7,9 +20,10 @@ export default ({ bodyHTML }: { bodyHTML: string }) => {
     <meta name="viewport" content="width=device-width,minimum-scale=1">
     <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
     <script async src="https://cdn.ampproject.org/v0.js"></script>
+    <style amp-custom>${resetCSS}${css}</style>
```
Comment (nicl) [inline frontend/amp/document.tsx]: I haven't looked in detail at what the resetCSS gives us here. I *think* it is worth keeping though - at the least it seems to clear out list styling which was useful here.

**Comment by nicl** [comment]
@SiAdcock - re pages, definitely happy to do this. I'm not entirely convinced we will serve pages dynamically in this way, unless we assume all pages require exactly the same data, but I can see it is helpful to mirror the existing structure for amp.

File: packages/rendering/server.ts (hunk 1 above)
Comment (AWare) [reply packages/rendering/server.ts]: I'm not really a fan of us calling react.createElement here. Can we have a seperate tsx file which composes the template and component into our response?

Comment (nicl) [reply packages/rendering/server.ts]: Okay gone with tsx'ing the file for now.

**Comment by SiAdcock** [comment]
@nicl Another thing about pages... they serve as entry points for browser bundles, ensuring we only serve the components needed for the page. There might be a more efficient way of defining entry points, but we'll know when we get more pages!

**Comment by nicl** [comment]
@SiAdcock ah cool thanks.

File: frontend/amp/components/Footer.tsx (hunk 3 above)
Comment (nicl) [reply frontend/amp/components/Footer.tsx]: Okay Zef says neutral 46 is a good alternative to the current prod choice so going for that!

Comment (nicl) [reply frontend/amp/components/Footer.tsx]: Anecdotally, it looks pretty similar, just a bit brighter.

**Pull request merged**